### ISSUE TITLE
Impossible to login with default language

### ISSUE THREAD

**Issue by apallier** [goharbor/harbor#9779]
**Expected behavior and actual behavior:**
It's impossible to login when default language is selected (French in my case), but it works when we choose/force a language at the top-right menu.

![image](https://user-images.githubusercontent.com/7916922/68368099-8dad5f00-0137-11ea-8c26-3d8b75447734.png)

**Steps to reproduce the problem:**
1. Remove all cookies and cache concerning Harbor site from your browser
2. Go to the Harbor front page => default language should be automatically detected (from browser info I guess), French in my case.
3. Try to login => error (see screenshot above)

It works if at step 2 I force language to English or other, and even if I force to the same language as the default one.

**Versions:**
- harbor version: 1.9.2 (problem seen juste after upgrade to 1.9.2)
- docker engine version: Docker version 18.03.1-ce, build 9ee9f40
- docker-compose version: docker-compose version 1.21.2, build a133471
- Chrome: 78.0.3904.87

**Additional context:**
Request when it doesn't work:
![image](https://user-images.githubusercontent.com/7916922/68370351-948aa080-013c-11ea-9641-f0a90c913996.png)

Request when it works:
![image](https://user-images.githubusercontent.com/7916922/68370376-a4a28000-013c-11ea-85eb-5d351a9f879f.png)

In /var/log/harbor/core.log when it doesn't work:

```
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] the request url is  /c/login#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] Handler crashed with error '_xsrf' argument missing from POST#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] /usr/local/go/src/runtime/panic.go:522#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] /harbor/src/vendor/github.com/astaxie/beego/context/context.go:80#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] /harbor/src/vendor/github.com/astaxie/beego/context/context.go:164#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] /harbor/src/vendor/github.com/astaxie/beego/controller.go:649#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] /harbor/src/vendor/github.com/astaxie/beego/router.go:788#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] /usr/local/go/src/net/http/server.go:2774#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] /usr/local/go/src/net/http/server.go:1878#033[0m
Nov  7 08:44:53 172.18.0.1 core[923]: 2019/11/07 07:44:53 #033[1;35m[C] [panic.go:522] /usr/local/go/src/runtime/asm_amd64.s:1337#033[0m
```

**Comment by reasonerjt** [comment]
@jwangyangls
I see the request is caryring xsrf token in the header
could we check why this fails?

**Comment by lssssssssss** [comment]
I have the same problem, when i try to balance harbor nodes with haproxy.
harbor nodes have common database and redis servers, type of balancing http/tcp - same result, in harbor.yml config defined external_url variable. Docker 19.03.5, harbor 1.9.2, and if i try to login on haproxy i have flapping error - "Handler crashed with error XSRF cookie", but if i login on the harbor node, its authorized me successfully

**Comment by KarstenSiemer** [comment]
This happens to me, too!
I do get `[panic.go:522] Handler crashed with error '_xsrf' argument missing from POST` all over the place and it only starts working when I have a harbor-lang cookie in place. And I only get this cookie when I change the language

**Comment by MatthiasHertel** [comment]
i have exactly the same issue deployed harbor via helm chart.

**Comment by akathimi** [comment]
+1

**Comment by akathimi** [comment]
@apallier We had almost the same error appear on a different scenario, we found that xsrf is enabled in on of the core config map(deployed on k8s), setting it to false solved the problem(as a workaround, this should be fixed by the harbor dudes).
the file path is /etc/core/app.conf 
`EnableXSRF = false
` 
hope this helps

**Comment by KarstenSiemer** [comment]
@reasonerjt now there seems to be a new problem after disabling xsrf:
```
Harbor:runtime error: invalid memory address or nil pointer dereference
Request Method: 	GET
Request URL: 	/c/oidc/callback?state=...&session_state=...&code=...
RemoteAddr: 	someIpAdress
Stack

/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/harbor/src/vendor/github.com/astaxie/beego/controller.go:579
/harbor/src/core/api/base.go:180
/harbor/src/core/controllers/oidc.go:151
/usr/local/go/src/reflect/value.go:447
/usr/local/go/src/reflect/value.go:308
/harbor/src/vendor/github.com/astaxie/beego/router.go:815
/usr/local/go/src/net/http/server.go:2774
/usr/local/go/src/net/http/server.go:1878
/usr/local/go/src/runtime/asm_amd64.s:1337

beego 1.9.0 (beego framework)

golang version: go1.12.12
```
I changed the IP and the URL a bit to make it more readable here.
Anyways this occasionally happens when trying to auth into the ui using oicd

**Comment by KarstenSiemer** [comment]
does anyone got an idea how to solve the mem nil pointer?
it essentially makes it impossible to login via oidc. But login via admin account still works.
Temporary fix is to delete the core and portal containers. After they come back up healthy and login works for some time

**Comment by fentas** [comment]
Getting the same error within a `kind` cluster trying to pull an image from harbor. Disabling `xsrf` solved it for as mentioned above. :cat2:

**Comment by AllForNothing** [comment]
This issue has been fixed for branches(release-1.9.0, release-1.10.0, release-2.0.0 and master )
